tests/e2e/bcd_consumer_integration_test.go

```diff
@@ -530,87 +523,65 @@ func (s *BCDConsumerIntegrationTestSuite) Test07ConsumerFPRewards() {
+	firstNonFinalizedBlock, err := s.babylonController.QueryCometBlock(firstNonFinalizedHeight)
 	s.NoError(err)
-	s.NotNil(activatedHeight)
-	submitHeight := activatedHeight.Height
+	s.NotNil(firstNonFinalizedBlock)
 
 	// get the babylon finality provider
 	babylonFp, err := s.babylonController.QueryFinalityProviders()
 	s.NoError(err)
 	s.NotNil(babylonFp)
-	babylonFpBIP340PK := bbn.NewBIP340PubKeyFromBTCPK(babylonFpBTCPK)
 
-	// Eventually find a block that is not timestamped
-	for {
-		s.T().Logf("Trying to equivocate for the block at height: %d", submitHeight)
-		// get the block at the submit height
-		submitHeightBlock, err := s.babylonController.QueryCometBlock(submitHeight)
-		s.NoError(err)
-		s.NotNil(submitHeightBlock)
-
-		randIdx := submitHeight - 1 // pub rand was committed from height 1-200
-
-		// submit finality signature
-		txResp, err := s.babylonController.SubmitFinalitySignature(
-			babylonFpBTCSK,
-			babylonFpBIP340PK,
-			randListInfo1.SRList[randIdx],
-			&randListInfo1.PRList[randIdx],
-			randListInfo1.ProofList[randIdx].ToProto(),
-			submitHeight)
-		s.NoError(err)
-		s.NotNil(txResp)
+	babylonFpBIP340PK := bbn.NewBIP340PubKeyFromBTCPK(babylonFpBTCPK)
+	randIdx := firstNonFinalizedHeight - 1 // pub rand was committed from height 1-100
 
-		// ensure the vote is eventually cast
-		var votes []bbn.BIP340PubKey
-		s.Eventually(func() bool {
-			votes, err = s.babylonController.QueryVotesAtHeight(submitHeight)
-			if err != nil {
-				s.T().Logf("Error querying votes: %v", err)
-				return false
-			}
-			return len(votes) > 0
-		}, time.Minute, time.Second*5)
-		s.Equal(1, len(votes))
-		s.Equal(votes[0].MarshalHex(), babylonFpBIP340PK.MarshalHex())
+	// submit finality signature
+	txResp, err := s.babylonController.SubmitFinalitySignature(
+		babylonFpBTCSK,
+		babylonFpBIP340PK,
+		randListInfo1.SRList[randIdx],
+		&randListInfo1.PRList[randIdx],
+		randListInfo1.ProofList[randIdx].ToProto(),
+		firstNonFinalizedHeight)
+	s.NoError(err)
+	s.NotNil(txResp)
 
-		// once the vote is cast, ensure the block is finalised
-		finalizedBlock, err := s.babylonController.QueryIndexedBlock(submitHeight)
-		s.NoError(err)
-		s.NotEmpty(finalizedBlock)
-		s.Equal(strings.ToUpper(hex.EncodeToString(finalizedBlock.AppHash)), submitHeightBlock.Block.AppHash.String())
-		s.True(finalizedBlock.Finalized)
-
-		// try to equivocate by submitting invalid finality signature
-		txResp, err = s.babylonController.SubmitInvalidFinalitySignature(
-			r,
-			babylonFpBTCSK,
-			babylonFpBIP340PK,
-			randListInfo1.SRList[randIdx],
-			&randListInfo1.PRList[randIdx],
-			randListInfo1.ProofList[randIdx].ToProto(),
-			submitHeight,
-		)
-		if err == nil {
-			s.NotEmpty(txResp)
-			s.T().Logf("Successfully submitted invalid finality signature")
-			break
-		}
-		// If the vote is rejected because the epoch is timestamped, continue to
-		// the next block
-		s.T().Logf("Failed to submit invalid finality signature: %v", err)
```

Comment:
And this. I guess it depends on the successful completion / status of the previous tests.

Reply:
Correct, after this refactoring submission is guaranteed to be successful